apps/camera/js/views/controls.js

```diff
@@ -148,19 +152,22 @@ module.exports = View.extend({
   },
 
   setMode: function(mode) {
+    mode = mode || 'picture';
     debug('set mode: %s', mode);
     this.set('mode', mode);
     this.switchPosition = this.switchPositions[mode];
     var ratio = { left: 0, right: 1 }[this.switchPosition];
-    this.updateSwitchPosition();
+    this.updateSwitchPosition(mode);
     this.setSwitchIcon(ratio);
     debug('mode set pos: %s', this.switchPosition);
   },
 
-  updateSwitchPosition: function() {
+  updateSwitchPosition: function(mode) {
```

Comment:
As this is an 'update' function I'd rather it gather the state it needs to update its stuff. 

```js
var mode = this.get('mode');
```